pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -0,0 +1,88 @@
+package component
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+
+	adapterCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+)
+
+// New instantiantes a component adapter
+func New(commonAdapter adapterCommon.AdapterContext, client kclient.Client) Adapter {
+	return Adapter{
+		Client:         client,
+		AdapterContext: commonAdapter,
+	}
+}
+
+// Adapter is a component adapter implementation for Kubernetes
+type Adapter struct {
```

Comment:
This struct is almost the same as the one in kubernetes/adapter.go

Can they be a common struct or is it required because of how we modularized the `Start()` over here? If so, can we rename it to `ComponentAdapter` instead of `Adapter`?

Reply:
Yeah, it should be separate because we're trying to make it modular. As for renaming it, if something imports this package then it is likely to be referenced as `component.Adapter` as is. Renaming it to ComponentAdapter would make it `component.ComponentAdapter` and the go linter will complain that it stutters. :)